Desktop 3.4.0 Release notes (#12971)
docker/docker.github.io: docker-for-mac/release-notes.md

{% include eula.md %}

## Docker Desktop 3.4.0
2021-06-07

[Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64){: .button .primary-btn}
[Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){: .button .primary-btn}

> ### New

Reviewer: Should we add a note on Volume Management and Compose canary roll out here?

Author: Ah yes, indeed. I’ll get some content from the blog post draft.

Reviewer 2: Yes. And what about the update notifications changes? (cc @dieucao)

Reviewer: Yes, totally forgot about that. Good point. I believe we should mention the change to update notification, but @dieucao's call :)

Author: I grabbed selected sentences from the blog post. I’m not sure the joyful tone is right for release notes, though. 😁

### Upgrades

- [Docker Engine 20.10.7](https://docs.docker.com/engine/release-notes/#20107)
- [Compose CLI v1.0.16](https://github.com/docker/compose-cli/releases/tag/v1.0.16)
- [Kubernetes 1.21.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.21.1)
- [containerd v1.4.6](https://github.com/containerd/containerd/releases/tag/v1.4.6)
- [runc v1.0.0-rc95](https://github.com/opencontainers/runc/releases/tag/v1.0.0-rc95)
- [Go 1.16.5](https://github.com/golang/go/releases/tag/go1.16.5)

### Bug fixes and minor changes

- Prevent `docker run` from hanging if inotify event injection fails. Fixes [docker/for-mac#5590](https://github.com/docker/for-mac/issues/5590).
- Fixed error showing stderr log in the UI. Fixes [docker/for-mac#5688](https://github.com/docker/for-mac/issues/5688).

> - Fix emulation of `riscv64`. Fixes [docker/for-mac#5699](https://github.com/docker/for-mac/issues/5699).

Reviewer: ```suggestion
- Fixed an issue which caused `riscv64` emulation to fail on Docker Desktop. Fixes [docker/for-mac#5699](https://github.com/docker/for-mac/issues/5699).
```

- Automatically reclaim space after deleting containers by deleting volumes and removing build cache.

> - Respect the bypass setting in the manual proxy configuration when pulling and pushing images. See [docker/for-mac#2467](https://github.com/docker/for-mac/issues/2467).

Reviewer: ```suggestion
- Docker Desktop now respects the bypass proxy setting through the **Manual proxy configuration** option when pulling and pushing images. See [docker/for-mac#2467](https://github.com/docker/for-mac/issues/2467).
```

Reviewer: Is it fixed both in the UI and the CLI?

Author: I don’t know, @djs55 ?

> - Fixed Docker Compose application removal from the UI which with custom file name other than `docker-compose.yml`. Fixes [docker/for-win#11046](https://github.com/docker/for-win/issues/11046)

Reviewer: How about " Fixed an issue which caused Docker Compose application to be removed from Docker Desktop if it contains a file name other than `docker-compose.yml`."

Author: It’s actually the other way around I believe: the application couldn’t be removed.

Author: Aha but indeed that sentence doesn’t make any sense 😀 
How about «Fixed deleting Docker Compose applications not removing from the UI when based on a file name different from `docker-compose.yml`.»?

Reviewer: Aah OK. Thanks for clarifying. I thought the application was automatically removed if it contained a custom file name :D 

Flipped the order to see if this reads better. "Docker Compose applications with file names other than `docker-compose.yml` can now be removed from Docker Desktop."

Author: Yes, that’s better. Thanks!

> - Expose the host CPU on Apple Silicon. Fixes [docker/for-mac#5681](https://github.com/docker/for-mac/issues/5681).

Reviewer: ```suggestion
- Docker Desktop now exposes the host CPU on Apple silicon. Fixes [docker/for-mac#5681](https://github.com/docker/for-mac/issues/5681).
```

- Avoid leaking open ports bound to privileged ports and specific IPs over engine Restart. Fixes [docker/for-mac#5649](https://github.com/docker/for-mac/issues/5649).
- Use `vpnkit` with `virtualization.framework` to fix connectivity issues with VPN clients such as Cisco AnyConnect.
- Fixed version number missing in update dialog window.

> - Fixed diagnostics sometimes not uploading correctly from the support dialog window.

Reviewer: ```suggestion
- Fixed an issue where the diagnostics were sometimes not uploaded correctly from the **Support** dialog.
```

## Docker Desktop 3.3.3
2021-05-06

> [Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/64133/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64){: .button .primary-btn}

Reviewer: ```suggestion
[Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/64133/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64)
```

Reviewer: Suggest removing the primary button here so we emphasize only the latest release

Author: Oh sorry, that answers my other question then. Thanks!

Author: Should we not add one for the Windows release notes as well, by the way?

Reviewer: Sorry, missed this somehow.

> Should we not add one for the Windows release notes as well, by the way?

Good question. We can add it only to the latest release for consistency.

> [Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/64133/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){: .button .primary-btn}

Reviewer: ```suggestion
[Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/64133/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64)
```

Author: Are you sure about that? Should I remove the buttons for all the old releases as well?

{% include eula.md %}

## Docker Desktop 3.4.0
2021-06-07

[Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64){: .button .primary-btn}
[Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){: .button .primary-btn}

### New

> In this release we’re introducing a new capability in Docker Desktop that helps you to create and delete volumes from Desktop’s Dashboard as well as to see which ones are In Use.

Reviewer: ```suggestion
**Volume Management**: Docker Desktop users can now create and delete volumes using the Docker Dashboard and also see which volumes are being used. For more information, see [Explore volumes](../desktop/dashboard.md#explore-volumes).

**Compose V2 beta**: Docker Desktop now includes the beta version of Compose V2, which supports the 'compose' command as part of the Docker CLI. For more information, see [Compose V2 beta](../compose/cli-command.md).

**Skip Docker Desktop updates**:  All users can now skip an update when they are prompted to install individual Docker Desktop releases. For more information, see [Docker Desktop updates](../docker-for-windows/install.md#updates).
```

Author: Thanks!

{% include eula.md %}

## Docker Desktop 3.4.0
2021-06-07

[Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64){: .button .primary-btn}
[Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){: .button .primary-btn}

### New

**Volume Management**: Docker Desktop users can now create and delete volumes using the Docker Dashboard and also see which volumes are being used. For more information, see [Explore volumes](../desktop/dashboard.md#explore-volumes).

**Compose V2 beta**: Docker Desktop now includes the beta version of Compose V2, which supports the 'compose' command as part of the Docker CLI. For more information, see [Compose V2 beta](../compose/cli-command.md).

**Skip Docker Desktop updates**: All users can now skip an update when they are prompted to install individual Docker Desktop releases. For more information, see [Docker Desktop updates](../docker-for-windows/install.md#updates).

### Deprecation

> - Docker Desktop no longer installs Notary, `docker trust` should be used for image signing.

Author: @usha-mandya ☝️

{% include eula.md %}

## Docker Desktop 3.4.0

> 2021-06-07

Reviewer: ```suggestion
2021-06-08
```

[Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64){: .button .primary-btn}
[Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){: .button .primary-btn}

### New

**Volume Management**: Docker Desktop users can now create and delete volumes using the Docker Dashboard and also see which volumes are being used. For more information, see [Explore volumes](../desktop/dashboard.md#explore-volumes).

**Compose V2 beta**: Docker Desktop now includes the beta version of Compose V2, which supports the 'compose' command as part of the Docker CLI. For more information, see [Compose V2 beta](../compose/cli-command.md).

**Skip Docker Desktop updates**: All users can now skip an update when they are prompted to install individual Docker Desktop releases. For more information, see [Docker Desktop updates](../docker-for-windows/install.md#updates).

### Deprecation

> - Docker Desktop no longer installs Notary, `docker trust` should be used for image signing.

Reviewer: ```suggestion
- Docker Desktop no longer installs Notary. You can now use [Docker Content Trust](../engine/security/trust.md) for image signing.
```

- [Docker Compose 1.29.2](https://github.com/docker/compose/releases/tag/1.29.2)
- [Docker Hub Tool v0.4.1](https://github.com/docker/hub-tool/releases/tag/v0.4.1)
- [Compose CLI v1.0.16](https://github.com/docker/compose-cli/releases/tag/v1.0.16)
- [Kubernetes 1.21.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.21.1)
- [containerd v1.4.6](https://github.com/containerd/containerd/releases/tag/v1.4.6)
- [runc v1.0.0-rc95](https://github.com/opencontainers/runc/releases/tag/v1.0.0-rc95)
- [Go 1.16.5](https://github.com/golang/go/releases/tag/go1.16.5)

### Bug fixes and minor changes

- Prevent `docker run` from hanging if inotify event injection fails. Fixes [docker/for-mac#5590](https://github.com/docker/for-mac/issues/5590).
- Fixed error showing stderr log in the UI. Fixes [docker/for-mac#5688](https://github.com/docker/for-mac/issues/5688).
- Fixed an issue which caused `riscv64` emulation to fail on Docker Desktop. Fixes [docker/for-mac#5699](https://github.com/docker/for-mac/issues/5699).
- Automatically reclaim space after deleting containers by deleting volumes and removing build cache.

> - Docker Desktop now respects the bypass proxy setting through the **Manual proxy configuration** option when pulling and pushing images. See [docker/for-mac#2467](https://github.com/docker/for-mac/issues/2467).

Reviewer 3: It's a more minor change:
```suggestion
- Docker Desktop now allows a blank HTTP proxy to be configured for the Docker engine, which will completely disable the internal HTTP proxy.  See [docker/for-mac#2467](https://github.com/docker/for-mac/issues/2467).
```

- Docker Compose applications with file names other than `docker-compose.yml` can now be removed from Docker Desktop. Fixes [docker/for-win#11046](https://github.com/docker/for-win/issues/11046)
- Docker Desktop now exposes the host CPU on Apple silicon. Fixes [docker/for-mac#5681](https://github.com/docker/for-mac/issues/5681).
- Avoid leaking open ports bound to privileged ports and specific IPs over engine Restart. Fixes [docker/for-mac#5649](https://github.com/docker/for-mac/issues/5649).
- Use `vpnkit` with `virtualization.framework` to fix connectivity issues with VPN clients such as Cisco AnyConnect.
- Fixed version number missing in update dialog window.
- Fixed an issue where the diagnostics were sometimes not uploaded correctly from the **Support** dialog.
- Fixed DNS entries for `*.docker.internal` and Kubernetes cluster reset after the VM IP changes.

> 

Reviewer 3: ```suggestion

### Known issues
- On Apple Silicon in native `arm64` containers, older versions of `libssl` in `debian:buster`, `ubuntu:20.04` and `centos:8` will segfault when connected to some TLS servers, for example `curl https://dl.yarnpkg.com`. The bug is fixed in newer versions of `libssl` in `debian:bullseye`, `ubuntu:21.04` and `fedora:35`.
```
